### PR TITLE
Fix for #81 to check if graphite server is defined in config

### PR DESCRIPTION
**Pull request by fatmcgav**
Fixes issue #81.

### CODE REVIEW

**Comment by colinPL** [comment]
This needs a rebase.

**Comment by fatmcgav** [comment]
@colinPL Will take a look now :)

**Comment by fatmcgav** [comment]
@colinPL Actually, looks like it's already been fixed here: https://github.com/puppetlabs/vmpooler/blob/fd90756b834dc6f31fa066abffd85312aed3f882/vmpooler#L12

Will give that a go tomorrow... :)

**Comment by fatmcgav** [comment]
I was still seeing nil class errors after pulling the latest changes in, so have tweaked the commit.

File: lib/vmpooler.rb [hunk 1]
```diff
@@ -33,7 +33,7 @@ def self.config(filepath='vmpooler.yaml')
     parsed_config[:config]['vm_checktime'] ||= 15
     parsed_config[:config]['vm_lifetime']  ||= 24
 
-    if parsed_config[:graphite]['server']
+    if defined? parsed_config[:graphite]['server']
```
Comment (colinPL) [inline lib/vmpooler.rb]: Do you know when `defined?` returns `nil` or `false`? Below it always returns a string, except when applied to an actually undefined variable (`k`).

``` ruby
irb(main):002:0> t = {graphite: {'server' => 'blah'}}
=> {:graphite=>{"server"=>"blah"}}
irb(main):003:0> defined? t
=> "local-variable"
irb(main):004:0> defined? k
=> nil
irb(main):005:0> defined? t[:graphite]['server']
=> "method"
irb(main):006:0> defined? t[:graphite]['asdf']
=> "method"
irb(main):007:0> defined? t[:graphite]
=> "method"
irb(main):008:0> defined? t[:graph]
=> "method"
```

Comment (colinPL) [inline lib/vmpooler.rb]: I think you want `parsed_config.has_key?(:graphite)`. That will check that the `:graphite` key even exists in the config, which will be `false` in your case.

**Comment by fatmcgav** [comment]
Reworked it... 

Checking for top-level `:graphite` key being set, but should it also check for or fail if `[:graphite][:server]` isn't set?

**Comment by colinPL** [comment]
@sschneid - thoughts?

**Comment by sschneid** [comment]
I think that checking for both `[:graphite]` and `[:graphite][:server]` makes sense since they'll both need to be set if Graphite support is going to work...

**Comment by sschneid** [comment]
I think that this fix is working, but I'm encountering a similar/related new (doh) error if I disable graphite while testing:

````
NoMethodError: undefined method `[]' for nil:NilClass
  (root) at vmpooler:11
````

I'm going to label this PR as 'blocked' until we can get a fix in so that this can be properly tested.

**Comment by colinPL** [comment]
Closing due to inactivity.

**Pull request closed without merging**